### ISSUE TITLE
Stylelint modifies jsx file even though no fix is reported or required

### ISSUE THREAD

**Issue by BPScott** [stylelint/stylelint#4485]
When using the css-in-js parser stylelint will reformat valid JS code that uses object shorthand even though no fix is reported or even required. This causes fun as stylelint produces code that conflicts with my eslint rule of [`object-shorthand: ['error', 'always']`](https://eslint.org/docs/rules/object-shorthand) - which ensures that object shorthands are preferable when possible (e.g. `{x: x}` should be rewritten as `{x}`.

Running stylelint --fix over the following code:

```jsx
const width = '100px';
const x = <div style={{width}}>Hi</div>
```

will cause it to be rewritten as

```jsx
const width = '100px';
const x = <div style={{width: width}}>Hi</div>
```

I would expect that my code is untouched.

## Reproduction

Create a new folder containing the following `package.json`:

```json
{
  "name": "stylelint-css-in-js",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {
    "stylelint": "12.0.0"
  },
  "stylelint": {
    "rules": []
  }
}
```

Add an `index.jsx` file:

```jsx
const width = '100px';
const x = <div style={{width}}>Hi</div>
```

Then install dependencies, and run stylelint over that file

```sh
npm install
node_modules/.bin/stylelint index.jsx --fix
```

Note that the file contents has now changed.

**Comment by fanich37** [comment]
Seems like it the same as #4449

**Comment by evilebottnawi** [comment]
Yes, it is duplicate of #4449
